python/python/lancedb/query.py

```diff
@@ -267,6 +267,16 @@ def to_list(self) -> List[dict]:
         """
         return self.to_arrow().to_pylist()
 
+    def to_batches(self, max_chunksize) -> list:
```

Comment:
We should name the item type here, which is a `RecordBatch`.

Also, we will likely change this in the future to return an iterator, rather than a list. I think many users will want to read batch-by-batch without having to load all of the data into memory at once. Therefore, we should just say `Iterable` in the type.
```suggestion
    def to_batches(self, max_chunksize) -> Iterable[pa.RecordBatch]:
```